### PR TITLE
ifaces:support exercising by required interface

### LINKED ISSUE #13434: interfaces: ApiCommand exercise by requiring interface, should check requiring interface is implemented in exercise guard.

### PR DESCRIPTION
**Pull request by robin-da**
This adds support to exercise an interface choice on a contract ID,
where the interface is required by one of the implemented interfaces of
the contract template.

Fixes #13434.

CHANGELOG_BEGIN
CHANGELOG_END

### CODE REVIEW

File: daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltin.scala [hunk 1]
```diff
@@ -1136,6 +1136,32 @@ private[lf] object SBuiltin {
...
+        case Right(ifaceSignature) if ifaceSignature.implements.contains(requiringIfaceId) =>
+          SBool(true)
+        case _ =>
+          throw SErrorDamlException(
+            IE.ContractDoesNotImplementRequiringInterface(
+              requiringIfaceId,
+              requiredIfaceId,
+              contractId,
+              actualTmplId,
+            )
```
Comment (sofiafaro-da) [inline daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltin.scala:1159]: Very nice! Could you also change `SBUnsafeFromRequiredInterface` so that it also throws `IE.ContractDoesNotImplementRequiringInterface`? Could be a separate PR.

File: daml-lf/language/src/main/scala/com/digitalasset/daml/lf/language/PackageInterface.scala [hunk 2]
```diff
@@ -317,7 +317,17 @@ private[lf] class PackageInterface(signatures: PartialFunction[PackageId, Packag
       case Right(interface) =>
         interface.fixedChoices.get(chName) match {
           case Some(choice) => Right(ChoiceInfo.Interface(choice))
-          case None => Left(LookupError(context, context))
+          case None => {
+            interface.requires.view
```
Comment (sofiafaro-da) [inline daml-lf/language/src/main/scala/com/digitalasset/daml/lf/language/PackageInterface.scala:321]: Can you add a TODO to improve this lookup, please, and link it to the issue https://github.com/digital-asset/daml/issues/11345

**Review by sofiafaro-da — APPROVED** [review]
Looks great, thank you!  I'm ok with adding tests in a separate PR.

File: daml-lf/language/src/main/scala/com/digitalasset/daml/lf/language/PackageInterface.scala (hunk 2 above)
Comment (robin-da) [reply daml-lf/language/src/main/scala/com/digitalasset/daml/lf/language/PackageInterface.scala:321]: done.

File: daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltin.scala (hunk 1 above)
Comment (robin-da) [reply daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltin.scala:1159]: I changed it.

**Review by remyhaemmerle-da — APPROVED** [review]
Thanks

File: daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/PhaseOne.scala [hunk 3]
```diff
@@ -352,7 +352,9 @@ private[lf] final class PhaseOne(
       case EUnsafeFromRequiredInterface(requiredIfaceId @ _, requiringIfaceId, cidExp, ifaceExp) =>
```
[the remark this answers is not in the document]
Comment (robin-da) [reply daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/PhaseOne.scala:352]: true, thanks!
